### PR TITLE
Box Storage: Add GetApplicationBoxes API and cucumber steps 

### PR DESCRIPTION
**Pull request by algochoi**
This PR adds the GetApplicationBoxes API endpoint in the v2 `algod` client and adds cucumber integration tests.

### CODE REVIEW

File: src/client/v2/algod/models/types.ts [hunk 1]
```diff
@@ -1066,6 +1066,52 @@ export class Box extends BaseModel {
   }
 }
 
+/**
```
Comment (algochoi) [inline src/client/v2/algod/models/types.ts:1069]: This is autogenerated from the spec

File: tests/cucumber/docker/run_docker.sh [hunk 2]
```diff
@@ -7,7 +7,7 @@ rm -rf test-harness
 rm -rf tests/cucumber/features
 
 # clone test harness
-git clone --single-branch --branch feature/box-storage https://github.com/algorand/algorand-sdk-testing.git test-harness
+git clone --single-branch --branch list-boxes https://github.com/algorand/algorand-sdk-testing.git test-harness
```
Comment (algochoi) [inline tests/cucumber/docker/run_docker.sh:10]: TODO: Revert before merge

File: tests/cucumber/steps/steps.js [hunk 3]
```diff
@@ -5419,6 +5426,29 @@ module.exports = function getSteps(options) {
...
+        boxes = splitAndProcessAppArgs(boxNames);
+      } else {
+        boxes = [];
+      }
+
+      const resp = await this.v2Client
+        .getApplicationBoxes(this.currentApplicationIndex)
+        .do();
+      const actualBoxes = new Set();
+      for (let i = 0; i < resp.boxes.length; i++) {
```
Comment (jdtzmn) [inline tests/cucumber/steps/steps.js:5443]: optional nit: could be a map for conciseness

File: tests/cucumber/steps/steps.js [hunk 4]
```diff
@@ -5419,6 +5426,29 @@ module.exports = function getSteps(options) {
     }
   );
 
+  Then(
+    'the current application should have the following boxes {string}.',
+    async function (boxNames) {
+      let boxes;
+      if (boxNames !== '') {
+        boxes = splitAndProcessAppArgs(boxNames);
```
Comment (jdtzmn) [inline tests/cucumber/steps/steps.js:5434]: nit: consider refactoring this logic into the `splitAndProcessAppArgs` method.

File: tests/cucumber/steps/steps.js [hunk 5]
```diff
@@ -5419,6 +5426,29 @@ module.exports = function getSteps(options) {
...
+      const resp = await this.v2Client
+        .getApplicationBoxes(this.currentApplicationIndex)
+        .do();
+      const actualBoxes = new Set();
+      for (let i = 0; i < resp.boxes.length; i++) {
+        const boxName = resp.boxes[i].name;
+        actualBoxes.add(Buffer.from(boxName, 'base64'));
+      }
+      const expectedBoxes = new Set(boxes.map((name) => Buffer.from(name)));
+      assert.deepStrictEqual(expectedBoxes, actualBoxes);
```
Comment (michaeldiamant) [inline tests/cucumber/steps/steps.js:5448]: @algochoi Leaving a visual reminder from our verbal to additionally assert returned array lengths match.

File: tests/cucumber/steps/steps.js (hunk 3 above)
Comment (algochoi) [reply tests/cucumber/steps/steps.js]: I think the suggestion looks better so I implemented it - thanks!

File: tests/cucumber/steps/steps.js (hunk 4 above)
Comment (algochoi) [reply tests/cucumber/steps/steps.js]: good point - done!

File: tests/cucumber/steps/steps.js (hunk 5 above)
Comment (algochoi) [reply tests/cucumber/steps/steps.js:5448]: done, thanks!

**Pull request merged**